lib/rows.js

```diff
@@ -274,6 +272,17 @@ function queryRowsOperationImpl(self, builtPlan, streamType, options, bindingArg
 return operation;
 }
 
+function createRowsEndpoint(builtPlan, options, graphqlQuery) {
+  let updateFunctions = new Set().add('write').add('lock-for-update').add('remove');
```

Comment:
We can't rely on this because it will only work for serialized plans. You should add a test that submits an Optic DSL statement that has write/remove/lockForUpdate in it - this logic won't work for it. 

Instead, per the design in the JIRA ticket, we need the user to tell us that they're doing an update, as we can't reliably determine that otherwise. That's the approach we're taking in the Java and Python clients too. The user can do that via an option - e.g:

    db.rows.query(planOrDslStatement, {"update": true})

To get data back, the user can do:

    db.rows.query(planOrDslStatement, {"update": true, "structure": "object"})